Its.Log.Monitoring.UnitTests/MonitoringTestDiscoveryTests.cs

```diff
@@ -391,23 +366,113 @@ public void when_a_tag_is_specific_the_casing_on_the_tag_does_not_matter()
-        public void properties_should_not_be_shows_as_tests()
+        public void public_void_methods_with_optional_parameters_should_be_shown_as_tests()
         {
             var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/").Result;
 
             response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("void_test_with_optional_parameters");
+        }
 
-            var json = response.Content.ReadAsStringAsync().Result;
+        [Test]
+        public void public_type_returning_methods_with_optional_parameters_should_be_shown_as_tests()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/").Result;
+
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("string_returning_test_with_optional_parameters");
+        }
+
+        [Test]
+        public void public_methods_with_non_optional_parameters_should_not_be_shown_as_tests()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/").Result;
+
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().NotContain("test_with_non_optional_parameters");
+        }
+
+        [Test]
+        public void public_methods_with_optional_parameters_have_the_option_parameter_set_to_the_default_value()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/" +
+                                              "string_returning_test_with_optional_parameters").Result;
+
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("bar");
+        }
+
+        [Test]
+        public void optional_test_parameters_can_be_set_with_a_query_string_parameter()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/" +
+                                              "string_returning_test_with_optional_parameters?foo=notbar").Result;
+
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("notbar");
+        }
+
+        [Test]
+        public void query_string_parameters_that_are_not_test_method_parameters_can_still_be_passed()
+        {
+            var response = apiClient.GetAsync("http://blammo.com/tests/staging/widgetapi/" +
+                                              "string_returning_test_with_optional_parameters?foo=notbar&apikey=iHaveNothingToDoWithYourTest").Result;
+
+            response.ShouldSucceed();
+            response.Content.ReadAsStringAsync().Result.Should().Contain("notbar");
+            response.Content.ReadAsStringAsync().Result.Should().NotContain("iHaveNothingToDoWithYourTest");
+        }
+
+        [Test]
+        public void when_a_parameter_of_the_wrong_type_is_supplied_a_useful_error_message_is_returned()

```

Comment:
Please include the word "then" in this kind of test name. It makes it easier to read.